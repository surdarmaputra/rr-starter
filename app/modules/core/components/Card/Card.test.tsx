import { render, screen, within } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from './Card';

describe('Card', () => {
  function renderComponent() {
    return render(
      <Card data-testid="Card-testid">
        <CardHeader>
          <CardTitle data-testid="CardTitle-testid">Card Title</CardTitle>
          <CardDescription data-testid="CardDescription-testid">
            Card Description
          </CardDescription>
        </CardHeader>
        <CardContent data-testid="CardContent-testid">
          <p>Card Content</p>
        </CardContent>
        <CardFooter data-testid="CardFooter-testid">
          <p>Card Footer</p>
        </CardFooter>
      </Card>,
    );
  }

  test('renders with data-testid', async () => {
    renderComponent();
    const card = await screen.findByTestId('Card-testid');
    const title = within(card).getByTestId('CardTitle-testid');
    within(title).getByText(/Card Title/);

    const description = within(card).getByTestId('CardDescription-testid');
    within(description).getByText(/Card Description/);

    const content = within(card).getByTestId('CardContent-testid');
    within(content).getByText(/Card Content/);

    const footer = within(card).getByTestId('CardFooter-testid');
    within(footer).getByText(/Card Footer/);
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
