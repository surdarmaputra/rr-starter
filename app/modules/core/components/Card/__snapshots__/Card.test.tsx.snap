// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders snapshot 1`] = `
<div>
  <div
    class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
    data-testid="Card-testid"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight"
        data-testid="CardTitle-testid"
      >
        Card Title
      </div>
      <div
        class="text-sm text-slate-500 dark:text-slate-400"
        data-testid="CardDescription-testid"
      >
        Card Description
      </div>
    </div>
    <div
      class="p-6 pt-0"
      data-testid="CardContent-testid"
    >
      <p>
        Card Content
      </p>
    </div>
    <div
      class="flex items-center p-6 pt-0"
      data-testid="CardFooter-testid"
    >
      <p>
        Card Footer
      </p>
    </div>
  </div>
</div>
`;
